go/vt/vtgate/evalengine/func.go

```diff
@@ -643,3 +645,48 @@ func (ta *typeAggregation) result() sqltypes.Type {
 	}
 	return sqltypes.VarChar
 }
+
+type builtinCeil struct{}
+
+func (builtinCeil) call(env *ExpressionEnv, args []EvalResult, result *EvalResult) {
+	inarg := &args[0]
+	argtype := inarg.typeof()
+	if inarg.isNull() {
+		result.setNull()
+		return
+	}
+
+	if sqltypes.IsIntegral(argtype) {
+		inarg.makeFloat()
+		result.setInt64(int64(math.Ceil(inarg.float64())))
```

Comment:
I don't see how this could possibly do anything! The result of converting an integral to a float64 will always be a round number, so calling `math.Ceil` on it will be a no-op. The whole roundtrip from integral -> float -> integral seems to be superfluous.

Have you found any corner cases that that roundtrip handles properly? From testing in the MySQL CLI, `CEIL` on an integer looks like a noop -- it always returns the same integer.